PWGHF/D2H/Tasks/taskDs.cxx

```diff
@@ -248,25 +306,45 @@ struct HfTaskDs {
   /// Fill histograms of quantities for the KKPi daugther-mass hypothesis
   /// \param candidate is candidate
   /// \param dataType is data class, as defined in DataType enum
-  template <CentralityEstimator centDetector, bool useMl, typename T1>
+  template <CentralityEstimator centDetector, bool useMl, typename Coll, typename T1>
   void fillHistoKKPi(const T1& candidate, DataType dataType)
   {
     auto pt = candidate.pt();
-    if constexpr (useMl) {
-      std::vector<float> outputMl = {-999., -999., -999.};
-      for (unsigned int iclass = 0; iclass < classMl->size(); iclass++) { // TODO: add checks for classMl size
-        outputMl[iclass] = candidate.mlProbDsToKKPi()[classMl->at(iclass)];
-      }
-      if constexpr (centDetector != CentralityEstimator::None) {
-        std::get<THnSparse_ptr>(histosPtr[dataType]["hSparseMass"])->Fill(hfHelper.invMassDsToKKPi(candidate), pt, evaluateCentrality<centDetector>(candidate), outputMl[0], outputMl[1], outputMl[2]);
+    if (dataType == DataType::Data) { // If data do not fill PV contributors in sparse
+      if constexpr (useMl) {
+        std::vector<float> outputMl = {-999., -999., -999.};
+        for (unsigned int iclass = 0; iclass < classMl->size(); iclass++) { // TODO: add checks for classMl size
+          outputMl[iclass] = candidate.mlProbDsToKKPi()[classMl->at(iclass)];
+        }
```

Comment:
This might cause a crash if you set `selectionFlagDs == 0` since candidates that do not pass the preselections have no entries in the vector of ML output scores (`candidate.mlProbDsToKKPi()`). 
It would be good to add a protection (i.e. if `candidate.mlProbDsToKKPi().size() == 0` continue). I never noticed before, but also `classMl` might be error prone because it should match what used in the selector (so it is a bit error prone)